src/components/SelectBox.tsx

```diff
@@ -1,35 +1,54 @@
-import React, { ChangeEventHandler } from "react";
+import React, { useContext } from "react";
 import styled from "styled-components";
-import { SelectBoxOption } from "../types/Restaurant";
+import { RestaurantContext } from "../App";
+import { Option } from "../types/Restaurant";
 
 interface SelectBoxProps {
-  options: SelectBoxOption[];
-  setState: ChangeEventHandler<HTMLSelectElement>;
+  name: Option;
+  options: { label: string; value: string }[];
 }
 
-class SelectBox extends React.Component<SelectBoxProps, {}> {
-
-  render() {
-    const { options, setState } = this.props;
-    return (
-      <Select onChange={setState}>
-        {options.map(option => (
-          <option value={option.value} key={option.value}>{option.label}</option>
-        ))}
-      </Select>
-    )
-  }
-}
+const SelectBox = (props: SelectBoxProps) => {
+  const { name, options } = props;
+  const { state, setState } = useContext(RestaurantContext);
+
+  const updateRestaurants = (value: string) => {
+    if (name === "sortBy")
+      setState({
+        ...state,
+        sortBy: value,
+        restaurants: state.restaurants
+          .slice()
+          .sort((a, b) => (a[value] > b[value] ? 1 : -1)),
+      });
+
+    if (name === "categorizeBy")
+      setState({
+        ...state,
+        categorizeBy: value,
+      });
+  };
+
+  return (
+    <Select name="" onChange={(event) => updateRestaurants(event.target.value)}>
```

Comment:
name이 공백이라면 제거해도 되는 속성 같네요!

```suggestion
    <Select name="" onChange={updateRestaurants}>
```

이렇게 선언하고 `updateRestaurants` 함수에 메모이제이션을 적용하면 어떨까요?